build/Dockerfile.rhel7

```diff
@@ -0,0 +1,13 @@
+FROM registry.svc.ci.openshift.org/ocp/builder:golang-1.13 AS builder
+WORKDIR /go/src/github.com/openshift/local-storage-operator
+COPY . .
+RUN make build-operator
+
+FROM registry.svc.ci.openshift.org/ocp/4.0:base
+COPY --from=builder /go/src/github.com/openshift/local-storage-operator/_output/bin/local-storage-operator /usr/bin/
+COPY manifests /manifests
+ENTRYPOINT ["/usr/bin/local-storage-operator"]
+LABEL com.redhat.delivery.appregistry=true
+LABEL io.k8s.display-name="OpenShift local-storage-operator" \
+      io.k8s.description="This is a component of OpenShift and manages local volumes." \
+        maintainer="Hemant Kumar <hekumar@redhat.com>"
```

Comment:
hmm what is the difference between these Dockerfiles and one in the top level repo?

Reply:
it is the same. Operator-SDK wants the docker files to be present in `build/Dockerfile`. We can remove the top-level docker files once this PR is merged after adjusting the ci config.

Reply:
All openshift images that @openshift/storage team ships has `Dockerfile` at top level and I would prefer that we keep this consistent.  Also I tested operator-sdk with latest version and while it is true that it generates a `Dockerfile` in `build` directory, it is only needed if you use `operator-sdk` tool to build images. So these files or directories are as far as I know not mandatory..

Reply:
okay removed the build dir